### ISSUE TITLE
(arcface torch) Partial FC speed is the same when setting sample_rate=0.1 and sample_rate=1

### ISSUE THREAD

**Issue by yongxinw** [deepinsight/insightface#1622]
I have observed that the under either sample rates, the trianing speeds are almost identical. Is changing the config.sample_rate=0.1 really triggering the partial fc operation? or am I missing something?

**Comment by anxiangsir** [comment]
please set num_classes to 2000000 or 10000000 to compare training speeds.

**Comment by yongxinw** [comment]
> please set num_classes to 2000000 or 10000000 to compare training speeds.

Thanks for your reply. I tried a model with 600000 classes but still didn't observe meaningful speed gains. I will try with 2000000 or 10000000 now.

I am wondering if setting the config.sample_rate=0.1 is the only change required to use Partial FC? Are there any other flags that need to be set in order to actually use the Partial FC?

**Comment by anxiangsir** [comment]
Yes, only changed `sample_rate`， In general, partial FC can save a lot of GPU memory and computation in the following cases:

1. When the number of indetities is greater than 2 millions.
2. In the model parallel training task of more than 32 GPUs. 

You can test 2M training tasks by only changing `cfg.num_classes` in the config.py file, and the following results can be reproduced.

![image](https://user-images.githubusercontent.com/31175974/125553161-1f28b6a6-c234-4239-bb2f-379d491a13cf.png)

**Comment by anxiangsir** [comment]
we updated a doc for training speed:

https://github.com/deepinsight/insightface/blob/master/recognition/arcface_torch/docs/speed_benchmark.md

**Comment by MyDecember12** [comment]
hello,have you tried to write npcface or CurricularFace in the form of PFC

**Comment by MyDecember12** [comment]
> hello,have you tried to write npcface or CurricularFace in the form of PFC



> we updated a doc for training speed:
> 
> https://github.com/deepinsight/insightface/blob/master/recognition/arcface_torch/docs/speed_benchmark.md

hello,have you tried to write npcface or CurricularFace in the form of PFC
